paper.tex

```diff
@@ -228,33 +250,127 @@ \subsection*{High-level model specification in YAML}
+
+The Demes specification precisely defines the required behaviour of
+parsers, and we also provide a reference implementation written
+in Python to resolve any potential ambiguities and to provide a
+helpful template for other implementations. We also provide an
+extensive suite suite of examples and the expected outputs. In addition,
+we have high-quality parser implementations in the Python and C
+languages (published under liberal open source licences),
+providing a solid foundation for the software ecosystem.
 
-\subsection*{Design rationale}
 
-This is an unstructured list of things we should mention in terms
-of why certain decisions were made
-
-\begin{itemize}
-    \item Recent developments in software engineering have favoured
-    static, declarative configuration formats. It feels like a good idea to
-    have a Turing complete configuration format, but it turns out not to be.
+\subsection*{Unambiguous low-level representation}
 
-    \item Drawing parameters from distributions. Why we are static.
-    % https://github.com/popsim-consortium/demes-paper/issues/3
+\jkcomment{Not sure if there's much point in this as a separate section any
+more? This could really be said in the previous section.}
+
+Parsers output the low-level description of Demes models,
+which is an explicit representation
+of the model where all parameters have been assigned values, and
+all data constraints have been checked. The output is formally defined
+as JSON, but in practise parsers output an object model that
+is suitable for the particular programming language, and
+that can be used directly by the implementing program.
+
+
+\subsection*{Scope of the specification}
+\jkcomment{This section is very rough.}
+
+% https://github.com/popsim-consortium/demes-paper/issues/3
+
+The Demes specification is by design simple and static---we wish to
+unambiguously describe a demographic model with fully specified
+and fixed parameters. This simplicity means that we cannot directly
+specify distributions or estimated confidence intervals
+for those parameters. While it is not difficult to imagine extending
+the specification in ways that would allows us to do this, it would
+greatly increase the complexity of parsers and would likely be of
+little practical use in any case. This is because the parameters of
+demographic models are typically tightly coupled, and cases in which
+distributions for different parameters can be simply described are rare.
+In this situation, the simplest way to describe estimated
+distribution is simply to list a large number samples from
+the posterior. While writing out a large number of Demes models in
+YAML format may seem inefficient, it can in fact be a compact
+way to describe these distributions.
+\jkcomment{TODO: add in a quick example here where we take an IM model
+and simulate a posterior distribution under some reasonable model. Report
+how much space it takes to represent a large number of samples from this
+distribution}
```

Comment:
Maybe mention that this will compress well/easily due to the information redundancy. Could also mention multi-document YAML files here. We should definitely hack some support for this into demes-python to get an idea of what multidoc files look like from an API perspective.

Digression: assuming no compression, I expect one multi-document file will be smaller than an equivalent number of separate files, because the space consumed by a file is an integer multiple of the filesystem's block size and practical models are very likely smaller than the filesystem's block size (often 4kb).

Reply:
Made new issue for this, #21